tools/cli/src/sourcemaps/add-sources.ts

```diff
@@ -117,92 +139,68 @@ export async function addSourcesToSourcemaps({ opts, logger, getHelpMessage }: C
-            .then(logTraceAsset('writing sourcemap'))
-            .then(writeSourceMap)
-            .then(logDebugAsset('sourcemap written'))
-            .thenErr(handleFailedAsset<AssetWithContent<RawSourceMap>>(logAssetBehaviorError(asset)))
-            .thenErr((error) => `${asset.name}: ${error}`).inner;
-
-    return AsyncResult.equip(find(...searchPaths))
-        .then(logDebug((r) => `found ${r.length} files`))
-        .then(map(logTrace((result) => `found file: ${result.path}`)))
-        .then(filter((t) => t.direct || matchSourceMapExtension(t.path)))
-        .then(map((t) => t.path))
-        .then(logDebug((r) => `found ${r.length} files for adding sources`))
-        .then(map(logTrace((path) => `file to add sources to: ${path}`)))
-        .then(opts['pass-with-no-files'] ? Ok : failIfEmpty('no sourcemaps found'))
-        .then(map(toAsset))
-        .then(map(loadSourceMapCommand))
-        .then(filterBehaviorSkippedElements)
-        .then(opts.force ? Ok : filterAssetsCommand)
-        .then(logDebug((r) => `adding sources to ${r.length} files`))
-        .then(map(logTrace(({ path }) => `file to add sources to: ${path}`)))
-        .then(
-            opts['pass-with-no-files']
-                ? Ok
-                : failIfEmpty('no sourcemaps without sources found, use --force to overwrite sources'),
-        )
-        .then(map(addSourceCommand))
-        .then(filterBehaviorSkippedElements)
-        .then(opts['dry-run'] ? Ok : map(writeSourceMapCommand))
-        .then(filterBehaviorSkippedElements).inner;
-}
-
-function doesSourceMapHaveSources(sourceProcessor: SourceProcessor) {
-    return function doesSourceMapHaveSources(asset: AssetWithContent<RawSourceMap>) {
-        return {
+    const addSourcesCommand = (asset: Asset) =>
+        pipe(
             asset,
-            result: sourceProcessor.doesSourceMapHaveSources(asset.content),
-        };
-    };
+            logTraceAsset('reading sourcemap'),
+            readSourceMapFromPathOrFromSource(sourceProcessor),
+            R.map(logDebugAsset('read sourcemap')),
+            R.map(logTraceAsset('adding source')),
+            R.map(addSourceToSourceMap(opts.force ?? false)),
+            R.map(logDebugAsset('source added')),
+            R.map(
+                opts['dry-run']
+                    ? Ok
+                    : flow(
+                          logTraceAsset('writing source and sourcemap'),
+                          writeAsset,
+                          R.map(logDebugAsset('wrote source and sourcemap')),
+                      ),
+            ),
+            R.mapErr((error) => `${asset.name}: ${error}`),
+            handleFailedAsset(logAssetBehaviorError(asset)),
+        );
+
+    const includePaths = normalizePaths(opts.include);
+    const excludePaths = normalizePaths(opts.exclude);
+    const { isIncluded, isExcluded } = await buildIncludeExclude(includePaths, excludePaths, logTrace);
+
+    return pipe(
```

Comment:
what happened to the rest commands available in the previous version? Why we don't need them anymore?

Reply:
ah ok, we have file per command now (feel free to correct me if I'm wrong)

Reply:
Umm, we always had file per command? `add-sources`, `process`, `upload`, and `run` are available.